webpack/assets/javascripts/react_app/components/SettingsTable/components/SettingCellInner.js

```diff
@@ -2,19 +2,32 @@ import React from 'react';
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
 import EllipsisWithTooltip from 'react-ellipsis-with-tooltip';
+import { useDispatch } from 'react-redux';
+
+import { setSettingEditing } from '../../SettingRecords/SettingRecordsActions';
+import useSettingModal from '../../SettingUpdateModal/useSettingModal';
 
 import { valueToString, hasDefault, inStrong } from '../SettingsTableHelpers';
 
 const SettingCellInner = props => {
-  const { setting, className, onEditClick, ...rest } = props;
+  const { setting, className, ...rest } = props;
 
   const cssClasses = classNames(className, {
     'editable-empty': !setting.value && setting.settingsType !== 'boolean',
     'masked-input': setting.encrypted,
   });
 
+  const { setModalOpen } = useSettingModal();
+  const dispatch = useDispatch();
+  if (!setting.readonly) {
+    rest.onClick = () => {
+      dispatch(setSettingEditing(setting));
+      setModalOpen();
+    };
+  }
```

Comment:
nit: This page will need a PF4 refactoring some day, so I'd say we need to keep it a little more explicit :)

```js
  const canClick = !setting.readonly;
  const openModal = () => {
    dispatch(setSettingEditing(setting));
    setModalOpen();
  };

  const field = (
    <span
      // with this, when `onClick` is undefined  react extracts it from the DOM.
      onClick={canClick ? openModal : undefined}
      {...rest}
      className={cssClasses}
    >
      {valueToString(setting)}
    </span>
  );
```